pilot/cmd/pilot-discovery/main.go

```diff
@@ -80,8 +80,10 @@ func init() {
 			bootstrap.KubernetesRegistry, bootstrap.ConsulRegistry, bootstrap.EurekaRegistry, bootstrap.CloudFoundryRegistry, bootstrap.MockRegistry))
 	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.CFConfig, "cfConfig", "",
 		"Cloud Foundry config file")
-	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.ClusterRegistriesDir, "clusterRegistriesDir", "",
-		"Directory for a file-based cluster config store")
+	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.ClusterRegistriesConfigmap, "clusterRegistriesConfigMap", "clusterregistry",
+		"ConfigMap map for clusters config store")
+	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.ClusterRegistriesNamespace, "clusterRegistriesNamespace", "istio-system",
+		"Namespace for ConfigMap which stores clusters configs")
```

Comment:
I think we can assume registry is in the same namespace with pilot.

Reply:
@costinm I still suggest to reconsider, saving all clusters admin kubeconf in the same name space is dangerous imho, but I will not fight for it ;) I will just get it on records that I was against it.